symphony-bdk-http/symphony-bdk-http-webclient/src/main/java/com/symphony/bdk/http/webclient/ApiClientWebClient.java

```diff
@@ -258,21 +262,30 @@ private void serializeApiClientBodyPart(String paramKey, ApiClientBodyPart bodyP
    *
    * @param authNames The authentications to apply
    */
-  private void updateParamsForAuth(String[] authNames, List<Pair> queryParams, Map<String, String> headerParams) {
+  private void updateParamsForAuth(String[] authNames, Map<String, String> headerParams) throws ApiException {
 
-    if (authNames == null) {
+    if (authNames == null && this.enforcedAuthenticationSchemes.isEmpty()) {
       return;
     }
-
+    authNames = withEnforcedSecurityScheme(authNames);
     for (String authName : authNames) {
       Authentication auth = this.authentications.get(authName);
       if (auth == null) {
         throw new RuntimeException("Authentication undefined: " + authName);
       }
-      auth.applyToParams(queryParams, headerParams);
+      auth.apply(headerParams);
     }
   }
 
+  private String[] withEnforcedSecurityScheme(String[] authNames) {
+    if (authNames == null) {
+      authNames = new String[0];
+    }
+
+    return Stream.concat(this.enforcedAuthenticationSchemes.stream(), Arrays.stream(authNames)).toArray(String[]::new);
+  }
```

Comment:
At some point it think that we will have to create an `AbstractApiClient` in the `symphony-bdk-http-api` module to avoid this kind of duplications